front/pages/api/apps/[user]/[sId]/runs/index.js

```diff
@@ -106,20 +110,59 @@ export default async function handler(req, res) {
       // console.log(inputDataset);
       // console.log(credentials);
 
+      const runRequestParams = {
+        run_type: "local",
+        specification: spec,
+        dataset_id: inputDataset ? inputDataset : null,
+        inputs: req.body.inputData ? req.body.inputData : null,
+        config: { blocks: config },
+        credentials,
+      };
+
+      if (req.body.stream) {
```

Comment:
Let’s fork here on “executeMode” and assume it means streaming, that’s one less parameter .

And let’s name that parameter “mode” and switch on it with for now 2 values “execute” and “design” maybe ?

“execute” mode should not update the app at all so you can just stream and return and keep the other branch unchanged